### PR TITLE
Azure docs

### PR DESCRIPTION
**Pull request by aguillenv**

### CODE REVIEW

File: docs/credentials.md [hunk 1]
```diff
@@ -10,5 +10,12 @@
 1. Create a service account for your GCP project as described [here](https://cloud.google.com/iam/docs/creating-managing-service-accounts#iam-service-accounts-create-console) with the [`Compute Engine Admin role`](https://cloud.google.com/compute/docs/access/iam#compute.admin).
 2. Create a service account key for your newly service account as described [here](https://cloud.google.com/iam/docs/creating-managing-service-account-keys) and export it as a JSON file.
 
+## Azure
 
+1. Create an Azure Active Directory application and it's Service Principal as described [here](https://docs.microsoft.com/en-us/azure/active-directory/develop/howto-create-service-principal-portal). After following the previous instructions you can copy your application (client) ID and your domain, also called directory (tenant) ID.
```
Comment (depombo) [inline docs/credentials.md:15]: going through it: `Application (client) ID` > application (client) ID and `Directory (tenant) ID` to make it easier to pattern match with the UI

Comment (depombo) [reply docs/credentials.md:15]: why did we call Directory (tenant) ID the `domain`? perhaps better to call it `directoryId`? The interface also had a URI which I thought was the `domain` for a second
Screen Shot 2021-03-02 at 8 51 59 AM

File: cli/src/deploy/mod.rs [hunk 2]
```diff
@@ -34,11 +34,21 @@ pub struct GCPCredentials {
   projectId: String,
 }
 
+#[allow(non_snake_case)]
+#[derive(Deserialize, Debug, Serialize)]
+pub struct AzureCredentials {
+  clientId: String,
```
Comment (depombo) [inline cli/src/deploy/mod.rs:40]: actually `appId` might make more sense here?

File: docs/credentials.md [hunk 3]
```diff
@@ -10,5 +10,12 @@
...
 
+2. Create a secret for the service principal created above as described [here](https://docs.microsoft.com/en-us/azure/active-directory/develop/howto-create-service-principal-portal#option-2-create-a-new-application-secret). Copy this secret immediately since you will not be able to retrieve it again later.
```
Comment (depombo) [inline docs/credentials.md:17]: another hint/cue given the UI: Copy this secret `Value` immediately
Screen Shot 2021-03-02 at 9 00 12 AM

File: docs/credentials.md (hunk 1 above)
Comment (aguillenv) [reply docs/credentials.md:15]: The nodejs client call it domain don't know why, but sure, we can call it `directoryId` or `tenantId` on the credentials file to make it easier

File: docs/credentials.md [hunk 4]
```diff
@@ -10,5 +10,12 @@
...
+
+3. For the subscription ID you can go to your [subscriptions page](https://portal.azure.com/#blade/Microsoft_Azure_Billing/SubscriptionsBlade) in Azure portal and get the ID.
+
+4. To be able to use Anycloud with Azure you will need to manage your subscription resource provider registration as described [here](https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/error-register-resource-provider#solution-3---azure-portal). You will need to register: `compute`, `network`, `storage` and `security`.
```
Comment (depombo) [inline docs/credentials.md:21]: per ui can we do `Microsoft.Compute` and so on even if it is redundant?
Screen Shot 2021-03-02 at 9 30 33 AM

Comment (aguillenv) [reply docs/credentials.md:21]: Yes

File: cli/src/deploy/mod.rs (hunk 2 above)
Comment (aguillenv) [reply cli/src/deploy/mod.rs:40]: Updated on: #16

**Review by depombo — APPROVED** [review]

**Pull request merged**